### PR TITLE
Fix #254: No way to figure out if radio button values have changed

### LINKED ISSUE #254: No way to figure out if radio button values have changed
### How to Reproduce
Change radio button value using the mouse, watch `formState`

### Expected behavior
When radio button value changes, the `formState.$dirty` becomes true

### Observed behavior
When radio button value changes, the `formState.$dirty` remains false

### PR DESCRIPTION
**Pull request by fdlk**
Fixes #254

#### Checklist
- [x] Functionality works & meets specifications
- [x] Code reviewed
- [x] Code unit/integration/system tested
- User documentation updated
- [x] Clean commits
- [x] No warnings during install
- Updated flow typing

### CODE REVIEW

File: src/components/field-types/RadioFieldComponent.vue [hunk 1]
```diff
@@ -79,6 +79,11 @@
     watch: {
       localValue () {
         this.$emit('input', this.localValue)
+
+        this.fieldState.$dirty = true
```
Comment (connoratrug) [inline src/components/field-types/RadioFieldComponent.vue]: why is this needed ? the vue-form should do this , if it does not please add a not why this work around is needed.

File: test/unit/specs/field-types/RadioFieldComponent.spec.js [hunk 2]
```diff
@@ -65,8 +65,12 @@ describe('RadioFieldComponent unit tests', () => {
...
   })
 
-  it('should emit an updated value on change', () => {
+  it('should emit an updated value and set fieldState on change', () => {
     wrapper.setData({localValue: '1'})
     expect(wrapper.emitted().input[0]).to.deep.equal(['1'])
+    expect(fieldState.$touched).to.equal(true)
+    expect(fieldState.$untouched).to.equal(false)
+    expect(fieldState.$dirty).to.equal(true)
+    expect(fieldState.$pristine).to.equal(false)
```
Comment (connoratrug) [inline test/unit/specs/field-types/RadioFieldComponent.spec.js]: please add check on the form state, possibly this is not updated now this is done outside of the vueform plugin

File: src/components/field-types/RadioFieldComponent.vue (hunk 1 above)
Comment (fdlk) [reply src/components/field-types/RadioFieldComponent.vue]: I understand why $touched does not become true. It triggers on blur and if you click a radio button it never becomes focused. The $dirty should become true when the value changes and I think this is probably a bug in the vue form library.

File: test/unit/specs/field-types/RadioFieldComponent.spec.js (hunk 2 above)
Comment (fdlk) [reply test/unit/specs/field-types/RadioFieldComponent.spec.js]: I've added an e2e test to check the browser integration

File: src/components/field-types/RadioFieldComponent.vue (hunk 1 above)
Comment (fdlk) [reply src/components/field-types/RadioFieldComponent.vue]: Added note

**Pull request merged**